data/mods/gen1/scripts.ts

```diff
@@ -693,13 +692,21 @@ export const Scripts: ModdedBattleScriptsData = {
 
 			// We now check attacker's and defender's stats.
 			let level = source.level;
-			let attacker = source;
-			const defender = target;
-			if (move.useTargetOffensive) attacker = target;
-			const atkType: StatIDExceptHP = (move.category === 'Physical') ? 'atk' : 'spa';
-			const defType: StatIDExceptHP = (move.defensiveCategory === 'Physical') ? 'def' : 'spd';
-			let attack = attacker.getStat(move.useSourceDefensiveAsOffensive ? defType : atkType);
+			const attacker = move.overrideOffensivePokemon === 'target' ? target : source;
+			const defender = move.overrideDefensivePokemon === 'source' ? source : target;
+
+			let atkType: StatIDExceptHP = move.category === 'Physical' ? 'atk' : 'spa';
+			if (move.overrideOffensiveStat) {
+				atkType = move.overrideOffensiveStat;
+			}
+			let defType: StatIDExceptHP = move.category === 'Physical' ? 'def' : 'spd';
+			if (move.overrideDefensiveStat) {
+				defType = move.overrideDefensiveStat;
+			}
```

Comment:
```suggestion
			const isPhysical = move.category === 'Physical';
			const atkType = move.overrideOffensiveStat || (isPhysical ? 'atk' : 'spa');
			const defType = move.overrideDefensiveStat || (isPhysical ? 'def' : 'spd');
```

Reply:
And with all _those_ improvements, this code can get a lot simpler, too!

Remember to also apply this to every other generation!

Reply:
Clever! I will make sure to roll these changes over to the other scripts as well. I will have time to make the commit in ~4 hours.